### ISSUE TITLE
Third argument of strncat should not be the length of src

### ISSUE THREAD

**Issue by maskit** [apache/trafficserver#3488]
gcc8 found a minor bug.

```
  CC       experimental/url_sig/url_sig.lo
experimental/url_sig/url_sig.c: In function 'urlParse':
experimental/url_sig/url_sig.c:397:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
         strncat(new_path_seg, "/", 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

It seems like the warning message is pointing out a misuse of `strncat`. Specifying the length of src on 3rd argument is the same as calling `strcat`.

Also, if you look into the block, it doesn't check that `new_path_seg` has a space for the "/". Since it won't work correctly if `new_path_seg` doesn't have enough space, we should probably call `strcat` after checking the length of `new_path_seg`.

**Comment by maskit** [comment]
Probably these three have the same issue.
```
experimental/url_sig/url_sig.c:441:7: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
       strncat(new_url, "/", 1);
       ^~~~~~~~~~~~~~~~~~~~~~~~
experimental/url_sig/url_sig.c:445:7: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
       strncat(new_url, "/", 1);
       ^~~~~~~~~~~~~~~~~~~~~~~~
experimental/url_sig/url_sig.c:450:7: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
       strncat(new_url, "/", 1);
       ^~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by SolidWallOfCode** [comment]
Fixed by #3731.
